beartype_test/a00_unit/a60_api/door/test_door.py

```diff
@@ -230,12 +226,12 @@ def muh_method(self):
             (MuhNutherThing, MuhThingP, False),
             (MuhThingP, MuhThing, False),
             # PEP 586-compliant type hints.
-            (Literal[1], int, True),
+            (Literal[7], int, True),
```

Comment:
have to change this from Literal[1] to something else because (oddly enough) it was breaking a check on the exception message over in codemain because of type caching in the typing module.  Whichever of `Literal[True]` and `Literal[1]` gets defined first becomes the permanent repr for _either_ of them! :joy:

Reply:
**ZOMG.** That's hilarious... and horrible! I'd just added this test case to the `test_door` submodule, too:

```python
        # Blame Guido.
        (bool, int, True),
```

Like, seriously. Law of Unintended Consequences strikes back.

`hash(1) == hash(True)` is such a festering pile of Bad Things™, because it all but guarantees hash collisions for anything mixing booleans and integers. How did Python 3.x never resolve that? But this...

> ...whichever of `Literal[True]` and `Literal[1]` gets defined first becomes the permanent repr for *either* of them! :joy:

Python: **AM I A JOKE TO YOU!?**

Guido: ![...heh](https://qph.cf2.quoracdn.net/main-qimg-f36f8215e0727969c4cce12af2c2977f-lq)